### PR TITLE
Improve template error reporting

### PR DESCRIPTION
**Pull request by zeripath**
There are multiple duplicate reports of errors during template rendering due to broken custom templates.

Unfortunately the error returned here is somewhat difficult for users to understand and it doesn't return the context of the error.

This PR attempts to parse the error returned by the template renderer to add in some further context including the filename of the template AND the preceding lines within that template file.

Ref #23274

### CODE REVIEW

**Comment by zeripath** [comment]
I've marked this as backport because it's precisely the kind of feature that is helpful to be backported.

File: modules/context/context.go [hunk 1]
```diff
@@ -228,6 +232,30 @@ func (ctx *Context) HTML(status int, name base.TplName) {
 			ctx.PlainText(http.StatusInternalServerError, "Unable to find status/500 template")
 			return
 		}
+		if _, ok := err.(texttemplate.ExecError); ok {
+			if groups := templateErr.FindStringSubmatch(err.Error()); len(groups) > 0 {
+				templateName, lineStr, posStr := groups[1], groups[2], groups[3]
+				target := "http"
```
Comment (zeripath) [inline modules/context/context.go]:
```suggestion
				target := ""
```

**Comment by lunny** [comment]
Could this display the root template name or the template dependency path when encountering a template error?

**Comment by zeripath** [comment]
> Could this display the root template name or the template dependency path when encountering a template error?

I can't report the template dependency path but I've at least added the a message to say if the issue is a subtemplate.

File: modules/context/context.go [hunk 2]
```diff
@@ -228,6 +232,34 @@ func (ctx *Context) HTML(status int, name base.TplName) {
...
+				}
+				line, _ := strconv.Atoi(lineStr) // Cannot error out as groups[2] is [1-9][0-9]*
+				pos, _ := strconv.Atoi(posStr)   // Cannot error out as groups[3] is [1-9][0-9]*
+				filename, filenameErr := templates.GetAssetFilename("templates/" + errorTemplateName + ".tmpl")
+				if filenameErr != nil {
+					filename = "(template) " + errorTemplateName
+				}
+				if errorTemplateName != string(name) {
+					filename += " (subtemplate of " + string(name) + ")"
+				}
```
Comment (delvh) [inline modules/context/context.go:250]: When does that happen?

**Review by delvh — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by jolheiser** [comment]
🎺 🤖

**Pull request merged**